Moving requirements out of use cases. Rewording metadata text to be more coherent. (#90)
SpencerDawkins/net-collab-rqmts: draft-kwbdgrr-tsvwg-net-collab-rqmts.md

  * Inspect client-to-server encrypted payload by network elements.

REQ-ISP-SCALE:
: The metadata and other state information that a router has to

> maintain for each additional media flow it handles should be kept

Reviewer: ```suggestion
maintain for each additional flow it handles should be kept
```

Reviewer: no need to know this is "media flow". Removing is consistent with the assumption in the draft that we don't care about the application type, etc.

Reviewer 2: Agree, just "flow" is better. (same for similar changes below)

The client authorization signal provides the means to convey the key
needed by the ISP to decrypt the metadata.

> REQ-CLIENT-DECIDES is satisfied by this in the client-to-network metadata

Reviewer: Not sure what is meant here.

Reviewer 2: Similar concern about "by this" in many of the updates (since we have definitions already for each requirement)

Author: By this, I meant the REQ-CLIENT-DECIDES is met by signaling Client flow authorization. The prev text shows Requirements under metadata definitions, which really didn't make sense given signaling these metadata is the way those requirements would be met. Maybe we can word it better to say 

"REQ-CLIENT-DECIDES is met by signaling Client Flow Authorization as part of client-to-network signal"?

Same line of thought for all other items as well.

Per-packet priority or importance determines the drop priority of
a packet.

> REQ-PACKET-PRIORITY is satisfied by this in the server-to-network metadata.

Reviewer: Idem as above

Author: same line of thought as REQ-CLIENT-DECIDES. Updated the language to be more specific.

the exact delay tolerance values that allow an on-path observer to
perform traffic analysis.

> REQ-PACKET-DELAY is satisfied by this in the server-to-network metadata.

Reviewer: Idem as above

Author: same line of thought as REQ-CLIENT-DECIDES. Updated the language to be more specific.

mitigate traffic analysis, the sender might, for example, purposefully mis-mark
metadata in some packets, add some randomness to avoid recurrent traffic patterns, etc.

REQ-PRIVACY-ADDITIONAL and REQ-SIGNALING-AVOIDANCE are satisfied by
not revealing any information that could identify the application's identity, reason to signal,

> server identity and securing the metadata.

Reviewer: ```suggestion
server identity, and securing the metadata.
```

mitigate traffic analysis, the sender might, for example, purposefully mis-mark
metadata in some packets, add some randomness to avoid recurrent traffic patterns, etc.

> Requirements REQ-PRIVACY-ADDITIONAL and REQ-SIGNALING-AVOIDANCE are satisfied by

Reviewer: ```suggestion
REQ-PRIVACY-ADDITIONAL and REQ-SIGNALING-AVOIDANCE are satisfied by
```

additional media flow it handles should be kept to a minimum or
eliminated altogether.

> REQ-ISP-SCALE is satisfied by this section.

Reviewer: Not sure we need to have this statement.

Author: I'll reword it and move the scalability to extended operational considerations and make ISP scale a requirement. That way the language is more consistent. Telling the metadata information should be kept to a minimum and then saying that is a requirement as well doesn't make sense

The principles outlined in {{?RFC8558}}, {{?RFC9049}} and {{?RFC9419}}
contain security considerations and are referenced in {{metadata-req}}.

Since the document focuses only on priorities within a flow

> (not specifying inter-flow priority), the entire requirement spec is not abusable by

Reviewer: ```suggestion
(not specifying inter-flow priority), the document does not induce concerns related to a specific
user or client declaring all flows or a subset of them as being more important. Such abuse concerns are thus not applicable.
```

> malicious traffic.

Reviewer: ```suggestion
```

themselves. However, the network cannot identify whether the
prioritized flow is legitimate or malicious.

> Such mechanism would satisfy REQ-CLIENT-VALIDATION.

Reviewer: Why are we adding this? We are not designing solutions.

Reviewer 3: also, 'the prioritized flow' smells like prioritizing *between* flows.  

There should be no harm in a flow that is marking some (or all) of its packets as important, even if it is doing so maliciously.

Author: Now reading it again, the whole Abuse and Constraints section seems to be in a "between flows" context than within the same flow. I agree with Dan as well, no harm in marking all packets important within the same flow. Can we remove the whole section?

Reviewer 3: because we are not prioritizing between flows, I believe we are immune from abuse.  Let's say a flow was all VoIP without any silence detection or anything, for example, so it is a solid stream of bits at a fixed rate.  Every packet could be marked High Priority.  Or Low Priority.  It doesn't matter.  If, while that VoIP is going on, the user adds a file transfer or the user adds screen sharing, then the priority begins to matter -- assuming the client considers the VoIP most important, the VoIP should get the highest priority among the 3 streams on that same flow (file transfer, VoIP, screen share).  A malicious sender would be noticed by the client both in the poorer user experience and in the packets being received which have incorrect priority markings.

Author: I agree Dan. Added text to reflect this into security considerations and removed Abuse section.

: Means to expose the signal independent of the application should be
considered. An example of such exposure is OS APIs.

REQ-NETWORK-SEEKS-LOAD-DOWN:

> : During detected reactive event, the network must implement a

Reviewer: ```suggestion
: During detected reactive events, the network implements a
```

reactive traffic policy to reduce or offload some of the traffic.
: This may involve utilizing alternative network attachments
available to the client (e.g., Wi-Fi).

REQ-CONTINUITY:
: Handover from one radio or router to another should continue to
provide same service level.

> : The network must deploy wireless routers closer to users to reduce link

Reviewer: ```suggestion
: The ISP deploys access routers closer to users to reduce link
```

Reviewer: This is deployment-specific. We can't mandate what an ISP can do.

Reviewer 2: Deploying network routers closer to the user/client does not resolve "continuity" requirement. What matters is that when the session is re-established (after handover), any negotiation of metadata is handling during the session re-establishment.

Author: ```
The general trend in wireless networks is to deploy the wireless
router closer to the user. This can help with low link latency but
can result in more frequent handovers as the user moves between
routers.
```

This was already there in the text. I just moved it under the requirement itself since it seems to be referring to a way it is done today (or at least reads so to me). Do we remove the whole statement altogether then?

Reviewer 3: it doesn't matter where or how things are deployed; what matters is the radio interface can provide feedback to the thing-doing-rate-limiting.  Which might be itself, but it still has to provide that feedback internally. It might be a different physical device or a VM or, in 5G, a virtualized function.  Doesn't matter at our level of requirements.

Author: Agreed! Removed those lines altogether and stated the requirement for what it is, succinctly.

Reviewer 2: @Sridhar: The trend to deploy wireless routers closer to the user is driven in 3GPP by capacity and latency concerns. That this results in more frequent handovers was to motivate the problem/requirement for continuity after handover. (i.e., unlike 4G-like centralized deployments, the lack of continuity for metadata would result in many users/sessions being affected). 
Just wanted to clarify: I don't see this as a big issue and what's in the draft is fine.

inbound/outbound bandwidth discussed in
{{?I-D.ietf-opsawg-teas-attachment-circuit}}).

Hints about resource-like metadata is bound by default to
the overall network attachment, not specific to a given application
or flow.

It is out of the scope of this document to discuss setups (e.g.,
3GPP PDU Sessions) where network attachments with Guaranteed Bit
Rate (GBR) for specific flows is provided.

Requirement: REQ-SCOPED-METADATA.

## Scalability {#scalability}

> There may be a large number of media flows handled by the server

Reviewer: ```suggestion
There may be a large number of flows handled by the server
```
